specifyweb/frontend/js_src/lib/components/QueryBuilder/Export.tsx

```diff
@@ -59,6 +65,33 @@ export function QueryExportButtons({
     });
   }
 
+  const [separator] = userPreferences.use(
+    'queryBuilder',
+    'behavior',
+    'exportFileDelimiter'
+  );
+
+  const selectedResults = resultsArray?.filter((item) => {
```

Comment:
this all should not be done on every render - wasteful performance wise
put it into a function and only call that function when the export button was pressed